controlplane/kubeadm/controllers/helpers_test.go

```diff
@@ -58,7 +59,9 @@ func TestReconcileKubeconfigEmptyAPIEndpoints(t *testing.T) {
 		recorder: record.NewFakeRecorder(32),
 	}
 
-	g.Expect(r.reconcileKubeconfig(ctx, clusterName, clusterv1.APIEndpoint{}, kcp)).To(Succeed())
+	result, err := r.reconcileKubeconfig(ctx, clusterName, clusterv1.APIEndpoint{}, kcp)
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(result).To(Equal(ctrl.Result{}))
```

Comment:
Since we do want the zero value of `ctrl.Result`
```suggestion
	g.Expect(result).To(BeZero())
```

Reply:
Same for the occurrences below.